simpleviewinc/keg-hub
# chore(retheme): Update dependencies

**Ticket**: [ZEN-614](https://jira.simpleviewtools.com/browse/ZEN-614)

## Goal

* Update the dependencies of retheme
* Fix any issues related to the dependency updates

## Updates
* `repos/re-theme/configs/babel.config.js`
  * Babel config fixes due to dep updates
* `repos/re-theme/configs/rollup.config.js`
  * Will now update the `react-native` exports to be `react-native-web` when building for web 
  * This will remove the need for the extra alias in `keg-core`, as they will already be correct
  * Added proptypes removal plugin, exactly the same as how keg-components works now
* `repos/re-theme/configs/webpack.config.js`
  * Minor update due to dep changes
* `repos/re-theme/package.json`
  * Updated all the deps to the latest versions where needed
* `repos/re-theme/src/**/*.stories.mdx`
  * Updated import due to deps updates

## Testing

### Test 1
* Run the command `keg retheme pack run retheme:zen-614-update-retheme`
* Navigate to [the retheme app](http://retheme-zen-614-update-retheme.local.keghub.io) in your browser
* Ensure the retheme app works as expected
* Click on all the navigation items, and ensure no errors are thrown for every page

### Test 2
* Run the command `keg components pack run components:zen-614-update-retheme`
  * This will run the keg-components app, with the retheme code changes of this PR
  * It includes no changes to keg-components
    * You will see a couple warnings about things that are out of date
    * This is expected and will be fixed in a seperate keg-components PR
* Navigate to [the keg-components app](http://components-zen-614-update-retheme.local.keghub.io) in your browser
* Ensure the the keg-components app works as expected
* Click on all the navigation items, and ensure no errors are thrown for every page

### Test 3
* Run this command `keg dpg run --package keg-core:ZEN-614-update-core --env prod`
  * Then navigate to [the keg-core app](http://core-zen-614-update-core.local.keghub.io/) in your browser
  * This keg-core image is using the changes from this PR 
  * Ensure every thing renders as expected
